drivers/gpio/gpio_mcux_lpc.c

```diff
@@ -111,49 +139,396 @@ static int gpio_mcux_lpc_read(struct device *dev,
+			int_flags = PINT_PinInterruptGetStatus(
+				config->pint_base, data->pint_id[pin]);
+			enabled_int =
+				(int_flags << pin) & data->pin_callback_enables;
+
+			PINT_PinInterruptClrStatus(config->pint_base,
+						   data->pint_id[pin]);
+
+			gpio_fire_callbacks(&data->callbacks, dev, enabled_int);
+		}
+	}
+}
+
+static u32_t get_free_isr(struct gpio_mcux_lpc_data *data)
+{
+	u32_t i;
+	u32_t isr;
+
+	for (i = 0; i < data->isr_list_idx; i++) {
+		if (data->isr_list[i] != -1) {
+			isr = data->isr_list[i];
+			data->isr_list[i] = -1;
+			return isr;
+		}
+	}
+
+	return -EINVAL;
+}
+
+/* Function configures INPUTMUX device to route pin interrupts to a certain
+ * PINT. PINT no. is unknown, rather it's determined from ISR no.
+ */
+static u32_t attach_pin_to_isr(u32_t port, u32_t pin, u32_t isr_no)
+{
+	u32_t pint_idx;
+	/* Connect trigger sources to PINT */
+	INPUTMUX_Init(INPUTMUX);
+
+	/* Code asumes PIN_INT values are grouped [0..3] and [4..7].
+	 * This scenario is true in LPC54xxx/LPC55xxx.
+	 */
+	if (isr_no < PIN_INT4_IRQn) {
+		pint_idx = isr_no - PIN_INT0_IRQn;
+	} else {
+		pint_idx = isr_no - PIN_INT4_IRQn;
+	}
+
+	INPUTMUX_AttachSignal(INPUTMUX, pint_idx,
+			      PIN_TO_INPUT_MUX_CONNECTION(port, pin));
+
+	/* Turnoff clock to inputmux to save power. Clock is only needed to make
+	 * changes. Can be turned off after.
+	 */
+	INPUTMUX_Deinit(INPUTMUX);
+
+	return pint_idx;
+}
+
+static void gpio_mcux_lpc_port_isr(void *arg);
+
+
+static int gpio_mcux_lpc_pin_interrupt_configure(struct device *dev,
+		unsigned int pin, enum gpio_int_mode mode,
+		enum gpio_int_trig trig)
+{
+	const struct gpio_mcux_lpc_config *config = dev->config->config_info;
+	struct gpio_mcux_lpc_data *data = dev->driver_data;
+	pint_pin_enable_t interruptMode = kPINT_PinIntEnableNone;
+	GPIO_Type *gpio_base = config->gpio_base;
+	u32_t port = config->port_no;
+	u32_t isr;
+	u32_t pint_idx;
+
+	/* Ensure pin used as interrupt is set as input*/
+	if ((mode & GPIO_INT_ENABLE) &&
+		((gpio_base->DIR[port] & BIT(pin)) != 0)) {
+		return -EINVAL;
+	}
+
+	switch (mode) {
+	case GPIO_INT_MODE_DISABLED:
+		interruptMode = kPINT_PinIntEnableNone;
+		break;
+	case GPIO_INT_MODE_LEVEL:
+		if (trig == GPIO_INT_TRIG_HIGH) {
+			interruptMode = kPINT_PinIntEnableHighLevel;
+		} else if (trig == GPIO_INT_TRIG_LOW) {
+			interruptMode = kPINT_PinIntEnableLowLevel;
+		} else {
+			return -ENOTSUP;
+		}
+		break;
+	case GPIO_INT_MODE_EDGE:
+		if (trig == GPIO_INT_TRIG_HIGH) {
+			interruptMode = kPINT_PinIntEnableRiseEdge;
+		} else if (trig == GPIO_INT_TRIG_LOW) {
+			interruptMode = kPINT_PinIntEnableFallEdge;
+		} else {
+			interruptMode = kPINT_PinIntEnableBothEdges;
+		}
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	/* First time calling this function routes PIN->PINT->INPUTMUX->NVIC */
+	if (data->pint_id[pin] == NO_PINT_INT) {
+		isr = get_free_isr(data);
+		if (isr == -EINVAL) {
+			/* Didn't find any free interrupt in this port */
+			return -EINVAL;
```

Comment:
Other drivers return `-EBUSY` meaning "Interrupt line required to configure pin interrupt is already in use."

Reply:
@mnkp done